### DISCUSSION TITLE
Integration with CUE

### DISCUSSION THREAD

**Discussion by apoorva-marisomaradhya** [grafana/grafonnet#33] [Q&A] [answered]
Hi,

I want to understand if there will be any Grafonnet integration with CUE going forward? Asking because we have seen a lot of reference to Grafonnet in https://github.com/grafana/grafana/discussions/39593.
We are currently investigating on methods to manage dashboards as code, it will be really helpful if you could provide some insight here.
Thank you.

**Comment by Duologic** [comment, marked as answer]
Good question. In linked discussion we've went all directions as we were finding our way around getting where we are today, this has certainly brought quite a bit of confusion.

Today we use CUE in grafana/grafana to define the schema's of all kinds of Grafana objects (dashboards, panels, datasources, ...) and we're still early and working hard on getting that into a better shape (this is a multi-team effort). With those CUE schema's we can do lots of interesting things, first of all improve maintainability of grafana/grafana itself but also generating client code.

Grafonnet is a first example of generating client code, to aid this we've build [grok](https://github.com/grafana/grok), it currently renders the CUE schemas into JSON schemas so we can use them in Grafonnet for a Jsonnet client. A I said, it's still early in the development and we're still figuring out how to best provide clients in other languages (think Golang, Typescript and perhaps CUE).

To answer the question, I don't think we'll provide a CUE integration in Grafonnet but we might come up with a separate CUE client in the future (no promisses). If you're keen on working with CUE yourself, I'd suggest have a look in the grafana/grafana repo for CUE files, note that these are not intended for public use, they change often without notice, don't expect support on it. :shrug:
